### PR TITLE
[WIP] Update Golang dependencies to Knative 1.0

### PR DESCRIPTION
**Pull request by pierDipi**
/hold

### CODE REVIEW

**Comment by pierDipi** [comment]
```
  Warning  Unhealthy           17s (x38 over 5m23s)  kubelet            Readiness probe failed: HTTP probe failed with statuscode: 400
  Normal   CloudEventObserved  7s                    observer-default   {"kind":"Received","error":"unknown Message encoding","httpHeaders":{"Accept":["*/*"],"Connection":["close"],"Host":["10.129.2.150:8080"],"User-Agent":["kube-probe/1.22"]},"origin":"10.129.2.1:37920","observer":"trans-pod","time":"2021-12-20T09:26:25.286826537Z","sequence":39}
```

**Comment by pierDipi** [comment]
@skonto we need to build and use v1.0 images first:

`trans-pod` is still using:
```
      image: 'quay.io/openshift-knative/recordevents:v0.26'
```

**Comment by pierDipi** [comment]
tldr; Eventing bump is tied to Golang bump

No need of additional commits in this PR, see https://github.com/openshift-knative/serverless-operator/pull/1347

/close
